Jenkinsfile

```diff
@@ -0,0 +1,63 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+/*
+ * Copyright 2019 Joyent, Inc.
+ */
+
+@Library('jenkins-joylib@v1.0.2') _
+
+pipeline {
+
+    agent {
+        label joyCommonLabels(image_ver: '1.6.3', pkgsrc_arch: 'i386')
+    }
+
+    options {
+        buildDiscarder(logRotator(numToKeepStr: '30'))
+        timestamps()
+    }
+
+    stages {
+        stage('check') {
+            steps{
+                sh('make check')
+            }
+        }
+        // avoid bundling devDependencies
+        stage('re-clean') {
+            steps {
+                sh('git clean -fdx')
+            }
+        }
+        stage('build image and upload') {
+            steps {
+                sh('''
+set -o errexit
+set -o pipefail
+
+export ENGBLD_BITS_UPLOAD_IMGAPI=true
+make print-BRANCH print-STAMP all release publish bits-upload''')
+            }
+        }
+        stage('agentsshar') {
+            // TODO: Consider complex handling of multiple branches
+            when {
+                branch 'master'
+            }
+            steps {
+                build(job:'agentsshar', wait: false)
+            }
+        }
```

Comment:
mackerel isn't in the agentsshar so we shouldn't have this part.

Reply:
yep, I changed this to 'mola' instead